lib/web_ui/lib/src/engine/text/canvas_paragraph.dart

```diff
@@ -644,11 +659,17 @@ class CanvasParagraphBuilder implements ui.ParagraphBuilder {
 
   @override
   void addText(String text) {
-    final EngineTextStyle style = _currentStyleNode.resolveStyle();
     final int start = _plainTextBuffer.length;
     _plainTextBuffer.write(text);
     final int end = _plainTextBuffer.length;
 
+    final EngineTextStyle style = _currentStyleNode.resolveStyle();
+    _updateCanDrawOnCanvas(style);
+
+    _spans.add(FlatTextSpan(style: style, start: start, end: end));
+  }
+
+  void _updateCanDrawOnCanvas(EngineTextStyle style) {
     if (_canDrawOnCanvas) {
       final ui.TextDecoration? decoration = style.decoration;
       if (decoration != null && decoration != ui.TextDecoration.none) {
```

Comment:
(this comment is meant for the line below)

After setting `_canDrawOnCanvas = false` we could immediately `return`. Then, together with the `if (!_canDrawOnCanvas)` suggestion above, we can remove all `if (_canDrawOnCanvas)`s (since we no longer need to populate `_spans`).

Reply:
Ah I see what you mean now. Disregard my previous comment.